File: lib/model/Device.dart
class Device {
  String battery;
  String deviceVersion;
  String id;
  String lastSyncTime;
  String type;

  Device(
      {this.battery,
      this.deviceVersion,
      this.id,
      this.lastSyncTime,
      this.type});

  Device.fromJson(Map<String, dynamic> json) {
    battery = json['battery'];
    deviceVersion = json['deviceVersion'];
    id = json['id'];
    lastSyncTime = json['lastSyncTime'];
    type = json['type'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['battery'] = this.battery;
    data['deviceVersion'] = this.deviceVersion;
    data['id'] = this.id;
    data['lastSyncTime'] = this.lastSyncTime;
    data['type'] = this.type;
    return data;
  }
}

File: android/app/src/main/java/com/example/smartsleep/model/Device.java
package com.example.smartsleep.model;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Device {

    @SerializedName("battery")
    @Expose
    public String battery;
    @SerializedName("batteryLevel")
    @Expose
    public Integer batteryLevel;
    @SerializedName("deviceVersion")
    @Expose
    public String deviceVersion;
    @SerializedName("features")
    @Expose
    public List<Object> features = null;
    @SerializedName("id")
    @Expose
    public String id;
    @SerializedName("lastSyncTime")
    @Expose
    public String lastSyncTime;
    @SerializedName("type")
    @Expose
    public String type;
    @SerializedName("mac")
    @Expose
    public String mac;

}